server/src/services/media.service.ts

```diff
@@ -206,8 +207,40 @@ export class MediaService {
             colorspace,
             quality: image.quality,
             processInvalidImages: process.env.IMMICH_PROCESS_INVALID_IMAGES === 'true',
+            angle: 0,
+            mirror: false
           };
 
+          if (asset.exifInfo?.orientation) {
+            switch (asset.exifInfo.orientation) {
+              case ExifOrientation.Horizontal:
+              case ExifOrientation.MirrorHorizontal:
+                imageOptions.angle = 0;
+                break;
+              case ExifOrientation.Rotate180:
+              case ExifOrientation.MirrorVertical:
+                imageOptions.angle = 180;
+                break;
+              case ExifOrientation.Rotate90CW:
+              case ExifOrientation.MirrorHorizontalRotate90CW:
+                imageOptions.angle = 90;
+                break;
+              case ExifOrientation.MirrorHorizontalRotate270CW:
+              case ExifOrientation.Rotate270CW:
+                imageOptions.angle = 270;
+                break;
+            }
+
+            if ([
+              ExifOrientation.MirrorHorizontal,
+              ExifOrientation.MirrorVertical,
+              ExifOrientation.MirrorHorizontalRotate90CW,
+              ExifOrientation.MirrorHorizontalRotate270CW,
+            ].includes(asset.exifInfo.orientation as ExifOrientation)) {
+              imageOptions.mirror = true;
+            }
+          }
+
```

Comment:
Let's put this in a method which returns { angle, mirror }

Reply:
Should I put it in media.service.ts? Or in some kind of ExifOrientationUtil class defined near ExifOrientation?

Reply:
The service is fine

Reply:
done